# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/Caches-inl.h
# Commit: 22854b710fbf
# Full Hash: 22854b710fbfb779ac6b19fe4c9ac2570189cec0
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2021-03-06 09:33:16
# Regressor Bug: 1689413
# File Overlap Count: 1
# Description:
#   Bug 1689413 part 6 - Move Realm* from ObjectGroup to BaseShape. r=jonco
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D106976
# ==============================================================================

diff -r 4531d5c25694 -r 22854b710fbf js/src/vm/Caches-inl.h
--- a/js/src/vm/Caches-inl.h	Fri Mar 05 19:10:06 2021 +0000
+++ b/js/src/vm/Caches-inl.h	Fri Mar 05 19:10:07 2021 +0000
@@ -50,11 +50,9 @@
   NativeObject* templateObj =
       reinterpret_cast<NativeObject*>(&entry->templateObject);
 
-  ObjectGroup* group = templateObj->group();
-
-  // If we did the lookup based on the proto we might have a group/object from a
+  // If we did the lookup based on the proto we might have a shape/object from a
   // different (same-compartment) realm, so we have to do a realm check.
-  if (group->realm() != cx->realm()) {
+  if (templateObj->shape()->realm() != cx->realm()) {
     return nullptr;
   }
 